src/main/scala/io/citrine/lolo/linear/LinearRegression.scala

```diff
@@ -76,19 +79,22 @@ case class LinearRegressionLearner(
       } else {
         Xw \ yw
       }
-      if (fitIntercept) {
-        (beta(0 to -2), beta(-1))
-      } else {
-        (beta, 0.0)
-      }
-    } getOrElse {
-      val totalWeight = weights.map(_.sum).getOrElse(numSamples.toDouble)
-      val mean = sum(yw) / totalWeight // weighted mean of training labels
-      if (fitIntercept) {
-        (DenseVector.zeros[Double](numFeatures - 1), mean)
-      } else {
-        (DenseVector.zeros[Double](numFeatures), mean)
-      }
+    } match {
+      case Success(beta) =>
+        if (fitIntercept) {
+          (beta(0 to -2), beta(-1))
+        } else {
+          (beta, 0.0)
+        }
+      case Failure(e) =>
+        logger.error(s"Encountered an exception solving normal equations: ${e.getLocalizedMessage}")
```

Comment:
Error log level is high. Is this situation significant enough to warrant automatic engineering attention, or should we log at a lower level and wait for customer feedback when it really should be working?

Reply:
I can move it to `warn`.

Reply:
That aligns with how we log internally in other places.